pyanaconda/ui/tui/spokes/time_spoke.py

```diff
@@ -145,3 +363,154 @@ def prompt(self, args = None):
     def apply(self):
         self.data.timezone.timezone = self._selection
         self.data.timezone.seen = False
+
+class NTPServersSpoke(NormalTUISpoke):
+    title = N_("NTP configuration")
+    category = LocalizationCategory
+
+    def __init__(self, app, data, storage, payload, instclass, time_spoke):
+        NormalTUISpoke.__init__(self, app, data, storage, payload, instclass)
+        self._time_spoke = time_spoke
+
+    def _summary_text(self):
+        """Return summary of NTP configuration."""
+        msg = _("NTP servers:")
+        if self._time_spoke.ntp_servers:
+            for status in format_ntp_status_list(self._time_spoke.ntp_servers):
+                msg += "\n%s" % status
+        else:
+            msg += _("no NTP servers have been configured")
+        return msg
+
+    def refresh(self, args=None):
+        NormalTUISpoke.refresh(self, args)
+
+        summary = self._summary_text()
+        self._window += [TextWidget(summary), ""]
+
+        _options = [_("Add NTP server")]
+        # only add the remove option when we can remove something
+        if self._time_spoke.ntp_servers:
+            _options.append(_("Remove NTP server"))
+        text = [TextWidget(m) for m in _options]
+
+        def _prep(i, w):
+            """ Mangle our text to make it look pretty on screen. """
+            number = TextWidget("%2d)" % (i + 1))
+            return ColumnWidget([(4, [number]), (None, [w])], 1)
+
+        # gnarl and mangle all of our widgets so things look pretty on screen
+        choices = [_prep(i, w) for i, w in enumerate(text)]
+
+        displayed = ColumnWidget([(78, choices)], 1)
+        self._window.append(displayed)
+
+        return True
+
+    def input(self, args, key):
+        try:
+            num = int(key)
+        except ValueError:
+            return key
+
+        if num == 1:
+            # add an NTP server
+            newspoke = AddNTPServerSpoke(self.app, self.data, self.storage,
+                                         self.payload, self.instclass, self._time_spoke)
+            self.app.switch_screen_modal(newspoke)
+            return INPUT_PROCESSED
+        elif self._time_spoke.ntp_servers and num == 2:
+            # remove an NTP server
+            newspoke = RemoveNTPServerSpoke(self.app, self.data, self.storage,
+                                            self.payload, self.instclass, self._time_spoke)
+            self.app.switch_screen_modal(newspoke)
+            return INPUT_PROCESSED
+        else:
+            # the user provided an invalid option number,
+            # just stay in the spoke
+            return INPUT_PROCESSED
+
+    def apply(self):
+        pass
+
+class AddNTPServerSpoke(EditTUIDialog):
```

Comment:
This (and RemoveNTPServerSpoke) should probably have the `indirect` property set to `True`, since they are not spokes you would ordinarily access via the hub. Probably TimeZoneSpoke as well now, since that's accessed via the general Time spoke.